dawn/src/dawn/Validator/IntegrityChecker.cpp

```diff
@@ -111,6 +111,12 @@ void IntegrityChecker::visit(const std::shared_ptr<iir::FieldAccessExpr>& expr)
     }
   }
 
+  auto dim = metadata_.getFieldDimensions(metadata_.getNameToAccessIDMap().at(expr->getName()));
+  if(!dim.K() &&
+     (expr->getOffset().hasVerticalIndirection() || expr->getOffset().verticalShift() != 0)) {
+    throw SemanticError("Attempting to read with vertical offset from 2 dimensional field!");
```

Comment:
```suggestion
    throw SemanticError("Attempting to read with vertical offset from horizontal field!");
```
One could confuse the spatial 1D, 2D, 3D nomenclature with the actual number of dimensions that a field has in unstructured, `Field[Edges]`: 1, `Field[Edges > Cells]`: 2, `Field[Edges > Cells, K]`: 3, and misinterpret this error message.